main.tf

```diff
@@ -1,13 +1,58 @@
-resource "random_integer" "example" {
+resource "kubernetes_service_account" "service_account" {
   count = module.this.enabled ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  metadata {
+    name        = var.deployment_name
+    namespace   = var.kubernetes_namespace
+    annotations = var.service_account_annotations
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
-}
+resource "kubernetes_deployment" "tfc_cloud_agent" {
+  count = module.this.enabled ? 1 : 0
+
+  metadata {
+    name      = var.deployment_name
+    namespace = var.kubernetes_namespace
+    labels    = var.labels
+  }
+  spec {
+    selector {
+      match_labels = var.selector_match_labels
+    }
+    replicas = var.replicas
+
+    template {
+      metadata {
+        labels      = var.labels
+        annotations = var.deployment_annotations
+      }
+      spec {
+        service_account_name            = kubernetes_service_account.service_account.0.metadata.0.name
+        automount_service_account_token = true
+        container {
+          image = var.agent_image
```

Comment:
Also support args... so we can do this:
https://www.terraform.io/docs/cloud/workspaces/agent.html#optional-configuration-single-execution-mode

Reply:
@osterman It looks like the `-single` arg is synonymous with the `TFC_AGENT_SINGLE` env var that we already support. In fact, I don't see any command line args that aren't already supported by the environment variables. Any thoughts on that?

Reply:
So, let's add args so the module will work automatically with newer versions of the cli, but default it to `[]`. 

Let's stick with using ENVs for all the known arguments.